### ISSUE TITLE
linter: false positive in `eslint-plugin-jest(valid-title)` rule

### ISSUE THREAD

**Issue by baseballyama** [oxc-project/oxc#8531]
### What version of Oxlint are you using?

0.15.6

### What command did you run?

oxlint -c=../../../.oxlintrc.json

### What does your `.oxlint.json` config file look like?

```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "error"
  },
  "plugins": ["typescript", "unicorn", "oxc", "security", "jsdoc", "n", "vitest"],
  "env": {},
  "settings": {},
  "rules": {
    "eqeqeq": ["error", "always", { "null": "ignore" }],
    "vitest/expect-expect": "off",
    "vitest/no-disabled-tests": "off",
    "vitest/no-conditional-expect": "off",
    "jest/no-standalone-expect": "off",
    "jest/no-export": "off",
    "no-document-cookie": "off",
    "no-unused-vars": "off"
  },
  "overrides": [
    {
      "files": ["*.svelte"],
      "rules": {
        "no-unused-vars": "off"
      }
    }
  ]
}
```

### What happened?

The type of `testCase` is `string`, but this error occurred.
I think we need to implement the configs provided by `eslint-plugin-vitest`.
https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/valid-title.md

![image](https://github.com/user-attachments/assets/83662059-cf94-4113-9794-b1702b304521)

**Comment by shulaoda** [comment]
In fact, `vitest/valid-title` also has this issue.
